src/Disks/ObjectStorages/MetadataStorageFromPlainRewritableObjectStorage.cpp

```diff
@@ -243,24 +244,21 @@ std::vector<std::string> MetadataStorageFromPlainRewritableObjectStorage::listDi
 {
     auto key_prefix = object_storage->generateObjectKeyForPath(path, "" /* key_prefix */).serialize();
 
-    RelativePathsWithMetadata files;
     auto absolute_key = std::filesystem::path(object_storage->getCommonKeyPrefix()) / key_prefix / "";
 
-    object_storage->listObjects(absolute_key, files, 0);
```

Comment:
This is the fix, the rest of the code is maintaining in-memory map with part filenames, so it can be used for listing objects.